scripts/scil_score_tractogram.py

```diff
@@ -0,0 +1,376 @@
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    p.add_argument('in_tractogram',
+                   help="Input tractogram to score")
+    p.add_argument('gt_bundles', nargs='+',
+                   help="Bundles ground truth(.[trk|tck], .nii or .nii.gz).")
+    g = p.add_argument_group('ROIs')
+    g.add_argument('--gt_endpoints', nargs='+',
+                   help="Bundles endpoints, both bundle's ROIs\
+                       (.nii or .nii.gz).")
+    g.add_argument('--gt_tails', nargs='+',
+                   help="Bundles tails, bundle's first ROI(.nii or .nii.gz).")
+    g.add_argument('--gt_heads', nargs='+',
+                   help="Bundles heads, bundle's second ROI(.nii or .nii.gz).")
+    p.add_argument('--dilate_endpoints',
+                   metavar='NB_PASS', default=1, type=int,
+                   help='Dilate masks n-times.')
+    p.add_argument('--gt_config', metavar='FILE',
+                   help=".json dict to specify bundles streamlines min, \
+                    max length and max angles.")
+    p.add_argument('--out_dir', default='gt_out/',
+                   help="Output directory")
+    p.add_argument('--wrong_path_as_separate', action='store_true',
+                   help="Separates streamlines that go outside of the ground \
+                        truth mask from true connections, outputs as \
+                        *_wpc.[tck|trk].")
+    p.add_argument('--remove_duplicate', action='store_true',
+                   help='Remove duplicate streamlines before scoring.')
+    p.add_argument('--remove_invalid', action='store_true',
+                   help='Remove invalid streamlines before scoring.')
+
+    add_json_args(p)
+    add_overwrite_arg(p)
+    add_reference_arg(p)
+    add_verbose_arg(p)
+
+    return p
+
+
+def extract_prefix(filename):
+    prefix = os.path.basename(filename)
+    prefix, _ = split_name_with_nii(prefix)
+
+    return prefix
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    assert_inputs_exist(parser, [args.in_tractogram] + args.gt_bundles)
+    assert_output_dirs_exist_and_empty(parser, args, args.out_dir,
+                                       create_dir=True)
+
+    if (args.gt_tails and not args.gt_heads) \
+            or (args.gt_heads and not args.gt_tails):
+        parser.error('Both --gt_heads and --gt_tails are needed.')
+    if args.gt_endpoints and (args.gt_tails or args.gt_heads):
+        parser.error('Can only provide --gt_endpoints or --gt_tails/gt_heads')
+    if not args.gt_endpoints and (not args.gt_tails and not args.gt_heads):
+        parser.error(
+            'Either input --gt_endpoints or --gt_heads and --gt_tails.')
+
+    if args.verbose:
+        logging.basicConfig(level=logging.INFO)
+
+    _, ext = os.path.splitext(args.in_tractogram)
+    sft = load_tractogram_with_reference(
```

Comment:
Would it be easier to remove invalid right after loading, rather than passing the argument to a function and just do it for fc/tc/fp/tp?

Reply:
Good point, fixed in 0434e4e